src/sentry/event_manager.py

```diff
@@ -110,19 +111,22 @@ def plugin_is_regression(group, event):
 
 
 def has_pending_commit_resolution(group):
-    return (
+    """
+    Checks that the most recent commit that fixes a group has had a chance to release
+    """
+    recent_group_link = (
         GroupLink.objects.filter(
             group_id=group.id,
             linked_type=GroupLink.LinkedType.commit,
             relationship=GroupLink.Relationship.resolves,
         )
-        .extra(
-            where=[
-                "NOT EXISTS(SELECT 1 FROM sentry_releasecommit where commit_id = sentry_grouplink.linked_id)"
-            ]
-        )
-        .exists()
+        .order_by("-datetime")
+        .first()
```

Comment:
We should use `first` method instead of `[0]` after filtering in other places as well

Reply:
Just note that we should only use `first` when we already have a sort specified, otherwise we might end up with queries that perform poorly.